### PR TITLE
Misc cleanups

### PR DESCRIPTION
**Pull request by sinkuu**

### CODE REVIEW

File: src/librustc_errors/emitter.rs [hunk 1]
```diff
@@ -528,9 +528,7 @@ impl EmitterWriter {
 
         // If there are no annotations or the only annotations on this line are
         // MultilineLine, then there's only code being shown, stop processing.
-        if line.annotations.is_empty() || line.annotations.iter()
-            .filter(|a| !a.is_line()).collect::<Vec<_>>().len() == 0
-        {
+        if line.annotations.iter().all(|a| a.is_line()) {
```
[the remark this answers is not in the document]
Comment (sinkuu) [reply src/librustc_errors/emitter.rs]: `Iterator::all` returns `true` for an empty iterator.

[the remark this answers is not in the document]
Comment (ljedrz) [reply src/librustc_errors/emitter.rs]: Ah, ok; nice :ok_hand:.

**Comment by ljedrz** [comment]
@sinkuu I doubt that your changes broke `asan`, so you might want to close and reopen this PR to trigger a re-test.

**Comment by sinkuu** [comment]
There was actually an error. The bot quoted irrevant parts of the log.

**Pull request merged**